lib/matplotlib/legend.py

```diff
@@ -1140,7 +1140,7 @@ def _get_legend_handles(axs, legend_handler_map=None):
         label = handle.get_label()
         if label != '_nolegend_' and has_handler(handler_map, handle):
             yield handle
-        elif (label not in ['_nolegend_', ''] and
+        elif (label != '' and label[0] != '_' and
```

Comment:
It may make sense to create a function `_is_active_label(label)` or something. For comparsion, see line 1160 below where the same condition is written completely different.

But for 3.6 this is probably OK.

Reply:
Could you simplify this by writing `not label.startwith('_')`?

Reply:
I copied the code below, which is indeed identical...  (had to check if an empty string was indeed False)